### PR TITLE
Adds "Hide results" option for polls

### PR DESCRIPTION
**Pull request by barsch**
This PR adds a new option called "Hide results" to polls which allow polls without seeing the results during the voting process which could influence the outcome of a poll.

Adding such option by the proposed way of overriding an application proved to be quite difficult as it spans over multiple applications (forum_conversation and forum_polls). I would really appreciate it if the feature makes into the main branch.

### CODE REVIEW

**Comment by barsch** [comment]
failing tests are as far I can see unrelated to this PR

**Comment by ellmetha** [comment]
> This PR adds a new option called "Hide results" to polls which allow polls without seeing the results during the voting process which could influence the outcome of a poll.

I understand the need to hide results _during_ the vote, but in this Pull Requests results are also hidden after the vote has been registered. Is this the intended behavior?

In the current state of this PR, only the author of the post can actually see the results of the poll. Shouldn't this behavior take into account actual user permissions? Eg. should forum admins or moderators be able to see the result of polls in all cases (regardless of the fact that results are hidden)?

**Comment by barsch** [comment]
> I understand the need to hide results _during_ the vote, but in this Pull Requests results are also hidden after the vote has been registered. Is this the intended behavior?

in my case: yes - the vote needs to be hidden during the vote and until a certain time (not necessary the end of the voting period) as the result will be published outside of the forum too - so we don't want voting trends to be visible before the actual end of the voting period which could influence the result again by animating people to counter vote etc ..

> In the current state of this PR, only the author of the post can actually see the results of the poll. Shouldn't this behavior take into account actual user permissions? Eg. should forum admins or moderators be able to see the result of polls in all cases (regardless of the fact that results are hidden)?

true - I'll implement it, once the issue above has been resolved

**Comment by ellmetha** [comment]
> in my case: yes - the vote needs to be hidden during the vote and until a certain time (not necessary the end of the voting period) as the result will be published outside of the forum too - so we don't want voting trends to be visible before the actual end of the voting period which could influence the result again by animating people to counter vote etc ..

I am fine with this behaviour, but administrators / moderators should be able to view the results I think.

File: machina/templates/machina/forum_conversation/forum_polls/poll_results.html [hunk 1]
```diff
@@ -1,5 +1,13 @@
 {% load i18n %}
+{% load forum_permission_tags %}
 
+{% get_permission 'can_edit_post' poll.topic.first_post request.user as user_can_edit_first_post %}
```
Comment (ellmetha) [inline machina/templates/machina/forum_conversation/forum_polls/poll_results.html:4]: 👍

**Review by ellmetha — APPROVED** [review]
Sorry for the delay! Otherwise, everything looks good to me. Thanks for contributing this! 😃

**Pull request merged**